controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -35,12 +35,13 @@ def _create_definition(self, host_definition):
                 host_definition.secret.namespace, host_definition.name))
             return
         response = self._define_host(host_definition)
-        host_definition.name = self._create_host_definition_if_not_exist(host_definition)
-        if response.error_message:
-            self._set_host_definition_status(host_definition.name, settings.PENDING_CREATION_PHASE)
-            self._create_event_to_host_definition(host_definition, response.error_message)
-        else:
-            self._set_host_definition_status(host_definition.name, settings.READY_PHASE)
+        host_definition_instance = self._create_host_definition_if_not_exist(host_definition)
+        if response.error_message and host_definition_instance:
```

Comment:
what if there is no host_definition_instance?

Reply:
that means that we couldn't create `hostdefinition` and we will throw the error into the logs [here](https://github.com/IBM/ibm-block-csi-driver/blob/ff28f837bd5404fd3a4f0051c5172e8c473ca9b5/controllers/servers/host_definer/kubernetes_manager/manager.py#L170-L173)

Reply:
and then what? what happen to the host e created it successfully?

Reply:
I am not sure what you mean, I think you missed a word

Reply:
sorry, what if the host has been created successfully on the storage? (no response.error_message but also host_definition_instance is None), what will be the outcome of the request?

Reply:
the host_definition_instance is None only if the host definer couldn't create `hostdefinition`, so if we will be able to define the host on the storage but not create a relevant `hostdefinition` on the cluster we will get errors in the logs